integration/deltaspike-data/impl/src/main/java/com/blazebit/persistence/impl/deltaspikedata/impl/builder/part/EntityViewQueryRoot.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.deltaspikedata.impl.builder.part;
+
+import com.blazebit.persistence.CriteriaBuilder;
+import com.blazebit.persistence.impl.deltaspikedata.impl.builder.EntityViewQueryBuilderContext;
+import com.blazebit.persistence.impl.deltaspikedata.impl.meta.EntityViewRepositoryComponent;
+import org.apache.deltaspike.data.impl.builder.MethodExpressionException;
+import org.apache.deltaspike.data.impl.meta.MethodPrefix;
+
+import static org.apache.deltaspike.data.impl.util.QueryUtils.splitByKeyword;
+
+/**
+ * Implementation is similar to {@link org.apache.deltaspike.data.impl.builder.part.QueryRoot} but was modified to
+ * work with entity views.
+ *
+ * @author Moritz Becker
+ * @since 1.2.0
+ */
+public class EntityViewQueryRoot extends EntityViewQueryPart {
```

Comment:
I think the where expression should be built eagerly and kept as final field. It doesn't seem like you actually require the CriteriaBuilder in the `EntityViewQueryBuilderContext` so please adapt that.

Reply:
Regarding the CriteriaBuilder: I need it in the context, what else should I apply the where expression on?

Reply:
Same goes for the order bys - I need to apply them to the CriteriaBuilder.

Reply:
You build the where expression string and collect order by expressions, this can be done without the a criteria builder during the building of the EntityViewQueryRoot. Applying the expressions on a crtieria builder should still happen, but you wouldn't re-inspect the method during runtime.